spikeinterface/preprocessing/preprocessing_tools.py

```diff
@@ -99,6 +95,58 @@ def get_spatial_interpolation_kernel(source_location, target_location, method='k
     if not force_extrapolate:
         interpolation_kernel[:, ~target_is_inside] = 0
 
-    
-
     return interpolation_kernel.astype(dtype)
+
+
+def get_kriging_kernel_distance(locations_1, locations_2, sigma_um, p):
+    """
+    Get the kriging kernel between two sets of locations.
+
+    Parameters
+    ----------
+
+    locations_1 / locations_2 : 2D np.array
+        Locations of shape (N, D) where N is number of
+        channels and d is spatial dimension (e.g. 2 for [x, y])
+    sigma_um : float
+        Scale paremter on  the Gaussian kernel,
+        typically distance between contacts in micrometers.
+    p : float
+        Weight parameter on the exponential function. Default
+        in IBL kriging interpolation is 1.3.
+
+    Results
+    ----------
+    kernal_dist : n x m array (i.e. locations 1 x locations 2) of
+                  distances (gaussian kernel) between locations 1 and 2.
+
+    """
+    dist = scipy.spatial.distance.cdist(locations_1, locations_2, metric='euclidean')
+    kernal_dist = np.exp(-(dist / sigma_um) ** p)
+    return kernal_dist
+
+
+def get_kriging_channel_weights(contact_positions, channel_indexes, sigma_um, p,
+                                weight_threshold=0.005):
+    """
+    Calculate weights for kriging interpolation. Weights below weight_threshold are set to 0.
+
+    Based on the interpolate_bad_channels() function of the International Brain Laboratory.
+
+    International Brain Laboratory et al. (2022). Spike sorting pipeline for the
+    International Brain Laboratory. https://www.internationalbrainlab.com/repro-ephys
+    """
+    weights = get_kriging_kernel_distance(contact_positions,
+                                          contact_positions[channel_indexes],
+                                          sigma_um,
+                                          p)
+    weights[channel_indexes, :] = 0
```

Comment:
I think this is the wring approach.
We have todo something like that
mask = ~np.in1d(channel_indexes, np.arange(num_channs))
someweights = get_kriging_kernel_distance(contact_positions[mask],
                                          contact_positions[channel_indexes],
                                          sigma_um,
                                          p)
And then  reconstruct the full vector

Reply:
I think I understand, just to check so we do here:
```
good_channel_only_mask = ~np.in1d(channel_indexes, np.arange(num_channs))
weights_of_good_channels_only_x_bad_channels= get_kriging_kernel_distance(contact_positions[good_channel_only_mask ],
contact_positions[channel_indexes],
sigma_um,
p)

then remove < threshold ... normalise ... remove < threshold ... return
```

So we only have matrix of distance weights between good channels and bad. Then in [interpolate_bad_channels.py](https://github.com/SpikeInterface/spikeinterface/pull/1025/files/53333b203e1e4c3b9d138f7d4ede41fee47dac0d#diff-c77514fdee3d39ae9ec959b339849ac58194e765087830fe2eaf44930a5ff032)
we do something like

```
interpolated_traces = traces_of_good_channels @ self.weights_of_good_channels_only_x_bad_channels
traces[:, self._bad_channel_indexes] = interpolated_traces 
```
and the purpose is to avoid filling the matrix with zeros and unecessary multiplication by zero when applying the weights?

Reply:
yes! I think it's a minor thing but it's a more elegant solution :) @JoeZiminski can you push the change?

Reply:
done